apps/render_cf.py

```diff
@@ -242,11 +237,50 @@ def validate_job_spec(job_type: str, job_spec: dict) -> None:
                 f'but should have {expected_param_fields}'
             )
 
+    for profile in job_spec['cost_profiles'].values():
+        if profile.keys() not in ({'cost_parameters', 'cost_table'}, {'cost'}):
+            raise ValueError(f'Cost definition for job type {job_type} has invalid keys: {profile.keys()}')
+
+        if 'cost_table' in profile:
+            if not isinstance(profile['cost_parameters'], list):
+                raise ValueError(
+                    f'Cost definition for job type {job_type} has invalid cost_parameters: Must be a list of strings.'
+                )
+
+            validate_cost_table(profile['cost_table'], job_type)
+
     for step in job_spec['steps']:
         if not step['image'].islower():
             raise ValueError(f'{job_type} has image {step["image"]} but docker requires the image to be all lowercase')
 
 
+def validate_cost_table(cost_table: dict | float, job_type: str) -> None:
+    if isinstance(cost_table, dict):
+        for key, value in cost_table.items():
+            if not isinstance(key, str) and not isinstance(key, int):
+                raise ValueError(
+                    f'Cost definition for job type {job_type} has invalid cost_table: '
+                    f'all cost_table keys must be strings or ints {type(key)}.'
+                )
+
+            return validate_cost_table(value, job_type)
+
+    elif isinstance(cost_table, float):
+        if not cost_table.is_integer():
+            raise ValueError(
+                f'Cost definition for job type {job_type} has invalid cost_table: '
+                'all cost_table costs must be whole numbers.'
+            )
+
+        return
+
+    else:
+        raise ValueError(
+            f'Cost definition for job type {job_type} has invalid cost_table: '
+            f'Cost table must be a nested dictionary of costs. {type(cost_table)}'
+        )
```

Comment:
Cool, I didn't know about `float.is_integer`. But we actually *don't* want to require that credit costs are integer values, we intentionally made them `Decimal` rather than `int` in case we ever want to support credit costs less than 1.
```suggestion
            validate_cost_table(value, job_type)

    elif not isinstance(cost_table, float):
        raise ValueError(
            f'Cost definition for job type {job_type} has invalid cost_table: '
            f'Cost table must be a nested dictionary of costs, got type {type(cost_table)}'
        )
```